app/services/search/query_builders/chat_channel_membership.rb

```diff
@@ -33,8 +34,14 @@ def initialize(params:)
 
       def build_queries
         @body[:query] = {}
-        @body[:query][:bool] = { filter: filter_conditions }
-        @body[:query][:bool][:must] = query_conditions if query_keys_present?
+        if !filter_conditions.select { |x| x[:term].key?(:channel_discoverable) }.empty?
```

Comment:
Rather than doing this complicated logic, here is a solution that I think would work better and requires no additional field or logic change here. 
- Remove channel_discoverable field
- If a channel is discoverable set `viewable_by` to `all`
```ruby
params[:viewable_by] = [<user_id>, "all"]
```
By default include multiple terms will be queried using an `OR`

Reply:
Hi @mstruve , i have optimized the above code to this. Please let me know what you think. 
`[      def build_queries
        @body[:query] = {}
        if @params[:user_id] == "all"
          @body[:query][:bool] = { filter: [{ terms: { status: %w[active joining_request] } }, { term: { channel_discoverable: true } }] }
          @body[:query][:bool][:must] = query_conditions
        else
          @body[:query][:bool] = { filter: filter_conditions }
          @body[:query][:bool][:must] = query_conditions if query_keys_present?
        end
      end](url)`